### PR TITLE
Allow creating presubmit job with restricted access to trigger it.

### PR DESCRIPTION
**Pull request by amwat**
Fixes [#5051](https://github.com/kubernetes/test-infra/issues/5051).

### CODE REVIEW

File: prow/config/jobs.go [hunk 1]
```diff
@@ -110,6 +110,9 @@ type Presubmit struct {
 
 	UtilityConfig
 
+	// The GitHub team id that is authorized to trigger this job.
+	AuthorizedTeam int `json:"authorized_team"`
```
Comment (BenTheElder) [inline prow/config/jobs.go]: I don't think this belongs in prowjobs. Presubmits should be unrestricted by default and a seperate config should restrict them by job name. We don't need to store this in every job.

**Comment by stevekuznetsov** [comment]
Is there a design document for this feature? How does it intersect with all of the other conditional job logic? Will anyone be allowed to re-run the job with `/test all` or `/retest`? Will re-triggering tests with a push re-trigger these jobs? How do we want to document this feature in `deck` and in the `k/community` repo?

**Comment by BenTheElder** [comment]
@stevekuznetsov just discussed this offline, pretty similar response. I'm not sure we have a valid enough use case for this, kinda hesitant to merge just on the principle of the feature.

If we don't trust a user enough to run any presubmits, fine. If we only trust them enough to run some, what exactly are we doing? Seems like this sort of functionality will create bad UX. To be fair, it was asked for in the past (scalability), but I think the solution so far has just been "manually triggered jobs" and so far that works. Nobody is spamming those recklessly (yet).

File: prow/github/fakegithub/fakegithub.go [hunk 2]
```diff
@@ -295,3 +295,28 @@ func (f *FakeClient) ListMilestones(org, repo string) ([]github.Milestone, error
 	}
 	return milestones, nil
 }
+
+func (f *FakeClient) GetTeamMembership(id int, user string) (*github.TeamMembership, error) {
```
Comment (fejta) [inline prow/github/fakegithub/fakegithub.go]: We've been preferring a fake web server over extending fakegithub, right?

**Comment by amwat** [comment]
Looping in @shyamjvs to decide whether this is still an ask.

**Comment by stevekuznetsov** [comment]
If it is, we should start with a design proposal sent to `kubernetes-sig-testing@` and determine if the larger community is OK with this direction before we attempt an implementation.

**Comment by fejta** [comment]
/lifecycle stale

Any further plans for this PR?

I'd recommend against adding complexity to solve hypothetical problems (people running more testing than the minimum required) until it proves to exist in reality.

**Comment by amwat** [comment]
Sounds fair, closing this now until we really need this.

**Comment by shyamjvs** [comment]
Sorry, I think I missed this earlier. We do need this as even though we've introduced large-scale scalability jobs as optional presubmits while ago, we haven't yet been able to announce this publicly (to the community) due to lack of proper access control in place. We won't be able to hold up too well if these jobs are triggered precariously.
